src/veil/ringct/anonwallet.cpp

```diff
@@ -3136,6 +3147,99 @@ bool AnonWallet::PickHidingOutputs(std::vector<std::vector<int64_t> > &vMI, size
     return true;
 }
 
+
+bool AnonWallet::GetRandomHidingOutputs(size_t nInputSize, size_t nRingSize, std::set<int64_t> &setHave, std::vector<CLightWalletAnonOutputData>& randomoutputs, std::string &sError)
```

Comment:
Again it would be ideal to have some shared code for use by these two functions, rather than duplicate a lot of tricky code. If possible.